rust_src/src/coding.rs

```diff
@@ -98,4 +99,31 @@ pub fn encode_file_name(fname: LispStringRef) -> LispStringRef {
     unsafe { c_encode_file_name(fname.into()) }.into()
 }
 
+/// Decode STRING which is encoded in CODING-SYSTEM, and return the result.
+/// Optional third arg NOCOPY non-nil means it is OK to return STRING
+/// itself if the decoding operation is trivial.
+/// Optional fourth arg BUFFER non-nil means that the decoded text is inserted in that buffer after point (point does not move). In this case, the return value is the length of the decoded text.
+/// This function sets `last-coding-system-used` to the precise coding system
+/// used (which may be different from CODING-SYSTEM if CODING-SYSTEM is not fully specified.)
+#[lisp_fn(min = "1")]
+pub fn decode_coding_string(string: LispObject, coding_system: LispObject, nocopy: Option<LispObject>, buffer: Option<LispObject>) -> LispObject {
+    code_convert_string(string, coding_system, buffer, false, nocopy.is_some(), false)
```

Comment:
```suggestion
    code_convert_string(string, coding_system, buffer, false, nocopy.is_not_nil(), false)
```

Reply:
`LispObject` is a C style `Option` it is either nil or a value. Wrapping it in `Option` is overkill. Same for the next function.